#!/usr/bin/env python
PACKAGE = "turtle_guard"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# 参数定义: name, type, level, description, default, min, max
gen.add("kp_linear", double_t, 0, "Linear control gain (P)", 1.5, 0.0, 5.0)
gen.add("kp_angular", double_t, 0, "Angular control gain (P)", 4.0, 0.0, 10.0)
gen.add("detection_radius", double_t, 0, "Detection Radius (m)", 3.0, 0.5, 8.0)
gen.add("wp_tolerance", double_t, 0, "Waypoint Tolerance (m)", 0.5, 0.1, 2.0)

exit(gen.generate(PACKAGE, "turtle_guard", "GuardConfig"))
